common_code/solver_cg_s_step.h

```diff
@@ -7,12 +7,12 @@
  * S-step CG according to Algorithm 2 in
  * https://research.nvidia.com/sites/default/files/pubs/2016-04_S-Step-and-Communication-Avoiding/nvr-2016-003.pdf
  */
+template <unsigned int n_steps = 1>
 class SolverCGSStep
 {
 public:
-  SolverCGSStep(SolverControl &cn, const unsigned int n_steps = 1)
+  SolverCGSStep(SolverControl &cn)
     : cn(cn)
-    , n_steps(n_steps)
     , times(6, 0.0)
   {
     AssertThrow(n_steps <= 8, ExcNotImplemented("Only up to 8 steps implemented"));
```

Comment:
Should this be removed?

Reply:
And further down in the file where I set up the C-style arrays of size [8], can you turn them into `[n_steps]`? Thank you.